cluster/engine.go

```diff
@@ -184,6 +185,14 @@ func (e *Engine) StartMonitorEvents() {
 func (e *Engine) ConnectWithClient(client dockerclient.Client) error {
 	e.client = client
 
+	// Use HTTP Client used by dockerclient to create engine-api client
```

Comment:
I wanted to do that, but there are some complications it causes. For instance, if we want to keep a similar workflow where `Connect()` initializes a client and `ConnectWithClient()` updates engine fields, then the signature for `ConnectwithClient()` will have to change to pass both `client` and `apiClient`. This is turn affects the unit tests (and the need to create a mock client for `engine-api`). Maybe I'm missing something, WDYT?